src/Symfony/Bridge/Twig/NodeVisitor/TranslationDefaultDomainNodeVisitor.php

```diff
@@ -49,19 +47,25 @@ public function enterNode(Node $node, Environment $env): Node
         }
 
         if ($node instanceof TransDefaultDomainNode) {
+            ++$this->nestingLevel;
+
             if ($node->getNode('expr') instanceof ConstantExpression) {
                 $this->scope->set('domain', $node->getNode('expr'));
 
                 return $node;
             }
 
-            $name = class_exists(AssignContextVariable::class) ? new AssignContextVariable(self::INTERNAL_VAR_NAME, $node->getTemplateLine()) : new AssignNameExpression(self::INTERNAL_VAR_NAME, $node->getTemplateLine());
-            $this->scope->set('domain', class_exists(ContextVariable::class) ? new ContextVariable(self::INTERNAL_VAR_NAME, $node->getTemplateLine()) : new NameExpression(self::INTERNAL_VAR_NAME, $node->getTemplateLine()));
-
             if (class_exists(Nodes::class)) {
+                $name = new LocalVariable(null, $node->getTemplateLine());
+                $this->scope->set('domain', $name);
+
                 return new SetNode(false, new Nodes([$name]), new Nodes([$node->getNode('expr')]), $node->getTemplateLine());
             }
 
+            $var = '__internal_trans_default_domain_'.$this->nestingLevel;
```

Comment:
I updated the implementation to rely on a counter instead

Reply:
@nicolas-grekas using a counter for the nesting level inside the current template does **not** solve the issue that this PR tries to fix, meaning that my comment in https://github.com/symfony/symfony/pull/57609#issuecomment-2437945576 is **not** solved and that Symfony 7.2 still breaks the case where both a parent template and its child template use `{% trans_default_domain %}`

Reply:
FTR, this was fixed in #58706